### PR TITLE
[ME-1041][Connector] improve refresh token handling

### PR DESCRIPTION
**Pull request by pedroandrade**
# Description

currently refresh token is only to refresh access tokens, but if the user is using a api token, we shouldn't refreshit, this one
improves the way that we deal with the access tokens.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: internal/api/api.go [hunk 1]
```diff
@@ -56,14 +66,17 @@ func WithVersion(version string) APIOption {
 
 type Border0API struct {
 	AccessToken string
+	Credentials *models.Credentials
 	Version     string
+	mutex       *sync.Mutex
```
Comment (adrianosela) [inline internal/api/api.go:71]: You don't need to initialize it below if you dont make it a pointer, and you can also have it itself be lockable, e.g.

```
type Border0API struct {
	sync.RWMutext // inheritss rw lock behavior

	AccessToken string
	Credentials *models.Credentials
	Version     string
}
```
then 

whenever you have a new `Border0API{}` it already has an initialized mutex, and you can use it as:

```
a.Lock()
defer a.Unlock()
```

File: internal/api/api.go [hunk 2]
```diff
@@ -352,3 +367,74 @@ func (a *Border0API) GetPoliciesBySocketID(socketID string) ([]models.Policy, er
...
+			case <-time.After(60 * time.Hour):
+				token, err := a.RefreshAccessToken()
+				if err != nil {
+					fmt.Println(err)
+
+					if err.Error() == "token is not valid to refresh" {
+						return err
+					}
+				}
+				a.mutex.Lock()
```
Comment (adrianosela) [inline internal/api/api.go:434]: prefer running as inline func w/ defer

```
func() {
        a.mutex.Lock()
        defer a.mutex.Unlock()

        a.Credentials = token
}()
```

File: internal/api/models/token.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+type Credentials struct {
+	AccessToken string `json:"access_token"`
+	ExpiresIn   int    `json:"expires_in"`
+	TokenType   string `json:"token_type"`
+}
+
+func (a *Credentials) ShouldRefresh() bool {
+	return a.TokenType == "User"
```
Comment (adrianosela) [inline internal/api/models/token.go:17]: Use const `CredentialsTypeUser`

**Review by adrianosela — APPROVED** [review]
Approved with nits

File: internal/api/api.go (hunk 1 above)
Comment (pedroandrade) [reply internal/api/api.go:71]: yup, but we are creating a global API lock, not sure if what I want, in my case I only need when I'm refreshing the token and updating the credentials

File: internal/api/models/token.go (hunk 3 above)
Comment (pedroandrade) [reply internal/api/models/token.go:17]: right :D

File: internal/api/api.go (hunk 1 above)
Comment (adrianosela) [reply internal/api/api.go:71]: sounds good. though then maybe the credentials object should have the lock in it.

File: internal/api/api.go (hunk 2 above)
Comment (adrianosela) [reply internal/api/api.go:434]: No, just best practice in go is to always use defer for unlocking right after locking.

File: internal/api/api.go (hunk 1 above)
Comment (pedroandrade) [reply internal/api/api.go:71]: that's a good point

File: internal/api/api.go (hunk 2 above)
Comment (pedroandrade) [reply internal/api/api.go:434]: cool, TIL :D

**Review by adrianosela — APPROVED** [review]

**Review by adrianosela — APPROVED** [review]

**Pull request merged**